packages/components/src/components/Button/Button.tsx

```diff
@@ -109,6 +115,31 @@ const getIcon = (iconName: IconNames, size: ButtonSizeOptions) => {
   return <Icon decorative icon={iconName} size={iconProps(size)} />;
 };
 
+const getButtonPadding = (
+  size: ButtonSizeOptions,
+  iconOnly: boolean
+): {
+  paddingBottom: SystemProp<keyof Theme["space"], Theme>;
+  paddingLeft: SystemProp<keyof Theme["space"], Theme>;
+  paddingRight: SystemProp<keyof Theme["space"], Theme>;
+  paddingTop: SystemProp<keyof Theme["space"], Theme>;
+} => {
+  if (iconOnly) {
+    return {
+      paddingBottom: size === "large" ? "space40" : "space30",
+      paddingTop: size === "large" ? "space40" : "space30",
+      paddingLeft: size === "large" ? "space40" : "space30",
+      paddingRight: size === "large" ? "space40" : "space30",
+    };
+  }
+  return {
+    paddingBottom: size === "large" ? "space40" : "space30",
+    paddingTop: size === "large" ? "space40" : "space30",
+    paddingLeft: size === "large" ? "space50" : "space40",
+    paddingRight: size === "large" ? "space50" : "space40",
```

Comment:
I was checking on Figma, and if the **small button** has an Icon (leading or trailing), it has only the left or right padding different from the others:

![image](https://user-images.githubusercontent.com/5320963/199513124-f40a8df0-37d2-46e1-b04b-a19035973257.png)

https://www.figma.com/file/0KtV7fzjug9hL3XOLzqn1l/%5B-WIP-%5D-%F0%9F%AA%90-Pluto-Components?node-id=149%3A2308